lisa/tests/staging/sched_android.py

```diff
@@ -0,0 +1,239 @@
+        self.boost = boost
+        self.prefer_idle = prefer_idle
+
+    @property
+    def cgroup_configuration(self):
+        return self.get_cgroup_configuration(self.plat_info, self.boost, self.prefer_idle)
+
+    @classmethod
+    def get_cgroup_configuration(cls, plat_info, boost, prefer_idle):
+        values = {
+                    'boost': boost,
+                    'prefer_idle': int(prefer_idle)
+                 }
+        return { 'name': 'lisa_test',
+                 'controller': 'schedtune',
+                 'values': values }
+
+    @classmethod
+    # Not annotated, to prevent exekall from picking it up. See
+    # SchedTuneBase.from_target
+    def from_target(cls, target, boost, prefer_idle, res_dir=None, ftrace_coll=None):
+        """
+        .. warning:: `res_dir` is at the end of the parameter list, unlike most
+            other `from_target` where it is the second one.
+        """
+        return super().from_target(target, res_dir, boost=boost,
+                prefer_idle=prefer_idle, ftrace_coll=ftrace_coll)
+
+    @classmethod
+    def _from_target(cls, target, res_dir, boost, prefer_idle, ftrace_coll=None):
+        plat_info = target.plat_info
+        rtapp_profile = cls.get_rtapp_profile(plat_info)
+        cgroup_config = cls.get_cgroup_configuration(plat_info, boost, prefer_idle)
+        cls._run_rtapp(target, res_dir, rtapp_profile, ftrace_coll, cgroup_config)
+
+        return cls(res_dir, plat_info, boost, prefer_idle)
+
+class SchedTuneBase(TestBundle):
+    """
+    Abstract class enabling the aggregation of ``SchedTuneItemBase``
+
+    :param test_bundles: a list of test bundles generated by
+        multiple ``SchedTuneItemBase`` instances
+    :type test_bundles: list
+    """
+    def __init__(self, res_dir, plat_info, test_bundles):
+        super().__init__(res_dir, plat_info)
+
+        self.test_bundles = test_bundles
+
+    @classmethod
+    def from_target(cls, target:Target, res_dir:ArtifactPath=None,
+            ftrace_coll:FtraceCollector=None) -> 'SchedTuneBase':
+        """
+        XXX: Warning if nothing is documented here ...
+        """
+        return super().from_target(target, res_dir, ftrace_coll=ftrace_coll)
+
+    @classmethod
+    def _from_target(cls, target, res_dir, ftrace_coll):
+        return cls(res_dir, target.plat_info,
+            list(cls._create_test_bundles(target, res_dir, ftrace_coll))
+        )
+
+    @classmethod
+    @abc.abstractmethod
+    def _create_test_bundles(cls, target, res_dir, ftrace_coll):
+        """
+        Collects and returns a ResultBundle per test item.
+        """
+        pass
+
+    @classmethod
+    def _create_test_bundle(cls, target, res_dir, ftrace_coll, item_cls, boost, pi):
+        """
+        Creates and returns a TestBundle for a given item class, and a given
+        schedtune configuration
+        """
+        item_dir = os.path.join(res_dir, 'boost_{}'.format(boost))
```

Comment:
Here you don't encode in the `pi` value... thus we can end up with some overwrites in case we test same boost values but different `pi` values...

nit-pick: should we use for consistency always the extended form, i.e. `s/pi/prefer_idle/`?

Reply:
Good catch, I missed that one. I'll fix it in the next version.